src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +11,32 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        Lock writeLock = lock.writeLock();
+        writeLock.lock();
+        try {
+            list.add(element);
+        } finally {
+            writeLock.unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        Lock readLock = lock.readLock();
+        readLock.lock();
+        try {
+            return list.get(index);
+        } finally {
+            readLock.unlock();
+        }
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        Lock readLock = lock.readLock();
+        readLock.lock();
+        try {
+            return list.size();
+        } finally {
+            readLock.unlock();
+        }
```

Comment:
In this 2 methods you create 2 objects of readLock, create field and reuse it in this 2 methods